### PR TITLE
Fixed compilation errors when targeting IOS.

### PR DESCRIPTION
**Pull request by nmoinvaz**

### CODE REVIEW

File: config_mac.c [hunk 1]
```diff
@@ -147,16 +153,21 @@ char *proxy_config_mac_get_bypass_list(void) {
         return NULL;
 
     // Get whether to exclude simple hostnames
-    bool exclude_simple_hostnames = get_cf_dictionary_bool(proxy_settings, kCFNetworkProxiesExcludeSimpleHostnames);
+    bool exclude_simple_hostnames = false;
```
Comment (sergio-nsk) [inline config_mac.c:156]:
```suggestion
    bool exclude_simple_hostnames = false;
    CFArrayRef exceptions_list = NULL;
```

File: config_mac.c [hunk 2]
```diff
@@ -147,16 +153,21 @@ char *proxy_config_mac_get_bypass_list(void) {
...
+#if !TARGET_OS_IPHONE
+    exclude_simple_hostnames = get_cf_dictionary_bool(proxy_settings, kCFNetworkProxiesExcludeSimpleHostnames);
     if (exclude_simple_hostnames)
         bypass_list_count++;
-
+#endif
     // Get exception list
-    CFArrayRef exceptions_list = CFDictionaryGetValue(proxy_settings, kCFNetworkProxiesExceptionsList);
+    CFArrayRef exceptions_list = NULL;
+#if !TARGET_OS_IPHONE
```
Comment (sergio-nsk) [inline config_mac.c:164]:
```suggestion
```

File: config_mac.c [hunk 3]
```diff
@@ -147,16 +153,21 @@ char *proxy_config_mac_get_bypass_list(void) {
...
 
     // Get whether to exclude simple hostnames
-    bool exclude_simple_hostnames = get_cf_dictionary_bool(proxy_settings, kCFNetworkProxiesExcludeSimpleHostnames);
+    bool exclude_simple_hostnames = false;
+#if !TARGET_OS_IPHONE
+    exclude_simple_hostnames = get_cf_dictionary_bool(proxy_settings, kCFNetworkProxiesExcludeSimpleHostnames);
     if (exclude_simple_hostnames)
         bypass_list_count++;
-
+#endif
```
Comment (sergio-nsk) [inline config_mac.c:161]:
```suggestion
```

File: net_adapter_linux.c [hunk 4]
```diff
@@ -10,7 +10,12 @@
 #include <unistd.h>
 #include <ifaddrs.h>
 #include <net/if.h>
-#include <net/if_arp.h>
+#ifdef HAVE_NET_IF_ARP_H
+#  include <net/if_arp.h>
```
Comment (sergio-nsk) [inline net_adapter_linux.c:14]: 3-space indent?

**Review by sergio-nsk — CHANGES_REQUESTED** [review]

File: net_adapter_linux.c (hunk 4 above)
Comment (nmoinvaz) [reply net_adapter_linux.c:14]: It is only two space indent. It is part of my lint configuration.

File: config_mac.c (hunk 1 above)
Comment (nmoinvaz) [reply config_mac.c:156]: I would prefer to keep the variables close to where they are being used and live with having two `#if/#endif`.

Comment (nmoinvaz) [reply config_mac.c:156]: I don't want to combine logic sections in order to merge two `#if/defs`.

**Review by sergio-nsk — APPROVED** [review]

**Pull request merged**